sycl/source/detail/devicelib/cmath_wrapper.cpp

```diff
@@ -0,0 +1,406 @@
+//==--- cmath_wrapper.cpp - wrappers for C math library functions ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifdef __SYCL_DEVICE_ONLY__
+#include "device_math.h"
+extern "C" {
+SYCL_EXTERNAL
+float __attribute__((weak)) scalbnf(float x, int n) {
```

Comment:
Please mention why we have __attribute__((weak)) here.

Reply:
OK, I will add comments here and add one execution test to show that the functions provided by users will have the highest priority and will override our "fallback" implementation.